### ISSUE TITLE
Don't copy ".github" directory when installing Magento using composer

### ISSUE THREAD

**Issue by hostep** [magento/magento2#22359]
### Preconditions (*)
1. Magento 2.2.4 - 2.3.1

### Steps to reproduce (*)
1. Install a Magento version somewhere between 2.2.4 and 2.3.1 using composer
2. Look in your installation directory

### Expected result (*)
1. Not finding a directory `.github`

### Actual result (*)
1. Finding a directory `.github`

### Discussion
This directory got added over here in the Github repo for various Github related things. So this should only be part of the files when cloning this repo, but not when setting up a new Magento project for a client.
The fact that it gets put in place, is because it gets delivered in the `magento/magento2-base` composer package. That package has an entry `extra > map > .github` which shouldn't be there.

Unfortunately the source code of the `magento/magento2-base` package isn't in this repo, so I can't create a PR to correct this. If somebody can let me know where we can find this, or if maybe some script is available which builds this package and if the source is publicly available, then I can take a look at fixing this.
Otherwise, it will be up to the Magento core devs to fix it, and maybe take a critical look at all the mappings in that package, maybe there are other things which aren't strictly needed there (`.travis.yml` maybe?) ...

**Comment by orlangur** [comment]
@hostep can't we achieve this with `.gitattributes`? https://stackoverflow.com/a/17069547

**Comment by ghost** [comment]
Hi @hostep thank you for you report, we are not able to accept issue that not related to the github repo, but we proceed this issue , in  internal ticket , and i'm will  notify you result, thanks for collaboration :+1:

**Comment by hostep** [comment]
@orlangur: it depends on how the `magento/magento2-base` module is getting build. Since that part of Magento Open Source isn't publicly available, unfortunately I can't say how it works.

I can however tell how the `.github` directory gets copied from the `magento/magento2-base` package into the project directory itself (I think Magento calls it "marshalling"), and that is done by the [magento-composer-installer plugin](https://github.com/magento/magento-composer-installer). The `magento/magento2-base` package specifies certain files/directories to be copied in its `composer.json` file:

```json
    "extra": {
        "map": [
            [
                ".github",
                ".github"
            ],
            [
                ".htaccess",
                ".htaccess"
            ],
            [
                ".htaccess.sample",
                ".htaccess.sample"
            ],
        ],
```

---

But I would like to see `.gitattributes` being used more often in every single Magento module though, we recently had a hosting company complaining about a lack of available inodes on the filesystem and they calculated that around 14000 files in the Magento code which gets put in place by composer are test/development related files. Those things don't make sense on a production environment, but I already [partly complained about that](https://github.com/magento/magento2/issues/5262) a long time ago.

```bash
$ find dev vendor/magento/*/Test -type f | wc -l
   14065
```

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-backlog-nazar 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-99243`, `MAGETWO-99244` were created

**Issue Available**: @engcom-backlog-nazar, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by hostep** [comment]
This is still happening on Magento 2.4.0, even though they actually cleaned up the `.travis.yml` file which I mentioned before as well.

Who can I talk to to make this happen? Making the source code responsible for this open source would help ...

**Comment by xmav** [comment]
@hostep We have internal ticket to fix that.

**Comment by hostep** [comment]
Looks like this was fixed in Magento 2.4.1, thanks!
